scanpy/tools/_umap.py

```diff
@@ -122,28 +125,48 @@ def umap(
         init_coords = check_array(init_coords, dtype=np.float32, accept_sparse=False)
 
     random_state = check_random_state(random_state)
-    n_epochs = 0 if maxiter is None else maxiter
     neigh_params = adata.uns['neighbors']['params']
     X = _choose_representation(
         adata, neigh_params.get('use_rep', None), neigh_params.get('n_pcs', None), silent=True)
-    # the data matrix X is really only used for determining the number of connected components
-    # for the init condition in the UMAP embedding
-    X_umap = simplicial_set_embedding(
-        X,
-        adata.uns['neighbors']['connectivities'].tocoo(),
-        n_components,
-        alpha,
-        a,
-        b,
-        gamma,
-        negative_sample_rate,
-        n_epochs,
-        init_coords,
-        random_state,
-        neigh_params.get('metric', 'euclidean'),
-        neigh_params.get('metric_kwds', {}),
-        verbose=settings.verbosity > 3,
-    )
+    if method == 'umap':
+        # the data matrix X is really only used for determining the number of connected components
+        # for the init condition in the UMAP embedding
+        n_epochs = 0 if maxiter is None else maxiter
+        X_umap = simplicial_set_embedding(
+            X,
+            adata.uns['neighbors']['connectivities'].tocoo(),
+            n_components,
+            alpha,
+            a,
+            b,
+            gamma,
+            negative_sample_rate,
+            n_epochs,
+            init_coords,
+            random_state,
+            neigh_params.get('metric', 'euclidean'),
+            neigh_params.get('metric_kwds', {}),
+            verbose=settings.verbosity > 3,
+        )
+    elif method == 'rapids':
```

Comment:
Maybe a warning would be enough here?

```suggestion
    elif method == 'rapids':
        metric = neigh_params.get('metric', 'euclidean')
        if metric != 'euclidean':
            raise ValueError(
                f'`sc.pp.neighbors` was called with `metric` {metric!r}, '
                "but umap `method` 'rapids' only supports the 'euclidean' metric."
            ) 
```